source/assets/js/playground/autocomplete.ts

```diff
@@ -0,0 +1,241 @@
+    validFor: identifier,
+  };
+}
+
+const moduleVariableCompletions = Object.freeze(
+  builtinModules.reduce(
+    (acc: {[k: string]: CompletionResult['options'] | []}, mod) => {
+      acc[mod.name] =
+        mod.variables?.map(variable => ({
+          label: `${mod.name}.${variable.name}`,
+          info: variable?.description,
+          type: 'variable',
+        })) || [];
+      return acc;
+    },
+    {}
+  )
+);
+
+const moduleFunctionsCompletions = Object.freeze(
+  builtinModules.reduce(
+    (acc: {[k: string]: CompletionResult['options'] | []}, mod) => {
+      acc[mod.name] =
+        mod.functions?.map(variable => ({
+          label: `${mod.name}.${variable.name}`,
+          apply: `${mod.name}.${variable.name}(`,
+          info: variable?.description,
+          type: 'method',
+          boost: 10,
+          validFor: identifier,
+        })) || [];
+      return acc;
+    },
+    {}
+  )
+);
+
+// Returns the list of built in modules that are included in the text.
+function includedBuiltinModules(state: EditorState) {
+  const text = state.doc.toString();
+  return moduleNames.filter(name => text.includes(`sass:${name}`));
+}
+
+// Completions for the namespaces of included built in modules.
+function builtinModuleNameCompletion(
+  context: CompletionContext
+): CompletionResult | null {
+  const nodeBefore = syntaxTree(context.state).resolveInner(context.pos, -1);
+  if (nodeBefore.type.name !== 'ValueName') return null;
+  // Prevent module name from showing up after `.`
+  if (nodeBefore.parent?.type.name === 'NamespacedValue') return null;
+  const includedModules = includedBuiltinModules(context.state);
+
+  const match = context.matchBefore(/\w+/);
+  if (!match) return null;
+
+  return {
+    from: match.from,
+    to: match.to,
+    options: includedModules.map(mod => ({
+      label: mod,
+      info: moduleDescriptions[mod],
+      type: 'namespace',
+      boost: 20,
+      validFor: identifier,
+    })),
+  };
+}
+
+// Completions for variables and functions for included built in modules.
+function builtinModuleItemCompletion(
+  context: CompletionContext
+): CompletionResult | null {
+  const nodeBefore = syntaxTree(context.state).resolveInner(context.pos, -1);
+  if (
+    ![nodeBefore.type.name, nodeBefore.parent?.type.name].includes(
+      'NamespacedValue'
+    )
+  )
+    return null;
+  const moduleNameMatch = context.matchBefore(moduleNameRegExp);
+
+  if (!moduleNameMatch) return null;
+  if (moduleNameMatch.from === moduleNameMatch.to && !context.explicit)
+    return null;
+
+  const includedModules = includedBuiltinModules(context.state);
+
+  const includedModFunctions = includedModules.flatMap(
+    mod => moduleFunctionsCompletions[mod]
+  );
+  const includedModVariables = includedModules.flatMap(
+    mod => moduleVariableCompletions[mod]
+  );
+
+  return {
+    from: moduleNameMatch.from,
+    to: moduleNameMatch.to,
+    options: [...includedModVariables, ...includedModFunctions],
```

Comment:
Won't this suggest all members for _all_ included modules, not just the one under the current namespace?